packages/flutter_tools/lib/src/isolated/native_assets/android/native_assets.dart

```diff
@@ -50,14 +50,25 @@ Future<Iterable<KernelAsset>> dryRunNativeAssetsAndroidInternal(
   const OSImpl targetOS = OSImpl.android;
 
   globals.logger.printTrace('Dry running native assets for $targetOS.');
-  final DryRunResult dryRunResult = await buildRunner.dryRun(
+  final BuildDryRunResult buildDryRunResult = await buildRunner.buildDryRun(
     linkModePreference: LinkModePreferenceImpl.dynamic,
     targetOS: targetOS,
     workingDirectory: projectUri,
     includeParentEnvironment: true,
   );
-  ensureNativeAssetsBuildSucceed(dryRunResult);
-  final List<AssetImpl> nativeAssets = dryRunResult.assets;
+  ensureNativeAssetsBuildDryRunSucceed(buildDryRunResult);
+  final LinkDryRunResult linkDryRunResult = await buildRunner.linkDryRun(
```

Comment:
why do we need to dry run the link phase?

Reply:
https://github.com/flutter/flutter/pull/148474#issuecomment-2125149031